presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -276,14 +277,24 @@ public void recordSubqueries(Node node, ExpressionAnalysis expressionAnalysis)
         return ImmutableList.of();
     }
 
-    public void addJoinInPredicates(Join node, JoinInPredicates joinInPredicates)
+    public void addJoinSideExpressions(Join node, JoinSideExpressions joinSideExpressions)
     {
-        this.joinInPredicates.put(node, joinInPredicates);
+        this.joinSideExpressions.put(node, joinSideExpressions);
     }
 
-    public JoinInPredicates getJoinInPredicates(Join node)
+    public JoinSideExpressions getJoinSideExpressions(Join node)
     {
-        return joinInPredicates.get(node);
+        return joinSideExpressions.get(node);
+    }
+
+    public void setPostJoinConjunct(Join node, Expression postJoinConjunct)
```

Comment:
The word "conjunct" is only relevant when describing terms of an AND expression. If anything, this should be called "postJoinPredicate".

Speaking of which, how is this different from the a regular "where" predicate? Any predicates that are applied after the join are semantically equivalent to being in the WHERE clause.